internal/metrics/metrics.go

```diff
@@ -0,0 +1,130 @@
+package metrics
```

Comment:
For a follow-up; depending where this all ends up in (i.e. if it's also included in packages that are imported externally), we could even consider having a build-tag to disable metrics at compile time.